tests/regression_test.py

```diff
@@ -0,0 +1,144 @@
+"""Tests for PROPKA 3.1"""
+import logging
+import os
+import re
+from pathlib import Path
+import pytest
+import pandas as pd
+import propka.lib
+import propka.molecular_container
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# Maximum error set by number of decimal places in pKa output as well as need
+# to make unmodified code work on WSL Ubuntu 18.04
+MAX_ERR = 0.01
+
+
+# This directory
+TEST_DIR = Path("tests")
+# Location for test PDBs
+PDB_DIR = Path("pdb")
+# Location for results for comparing output (allow running from tests/ and ../tests/)
+RESULTS_DIR = Path("tests/results")
+if not RESULTS_DIR.is_dir():
+    _LOGGER.warning("Switching to sub-directory")
+    RESULTS_DIR = Path("results")
+# Arguments to add to all tests
+DEFAULT_ARGS = []
+
+
+def get_test_dirs():
+    """Get locations of test files.
+
+    Returns:
+        dictionary with test file locations.
+    """
+    path_dict = {}
+    for key, path in [("pdbs", PDB_DIR), ("results", RESULTS_DIR)]:
+        test_path = TEST_DIR / path
+        if test_path.is_dir():
+            path_dict[key] = test_path
+        else:
+            test_path = path
+            if test_path.is_dir():
+                path_dict[key] = test_path
+            else:
+                errstr = "Can't find %s test files in %s" % (key, [TEST_DIR / path, path])
+                raise FileNotFoundError(errstr)
+    return path_dict
+
+
+def run_propka(options, pdb_path, tmp_path):
+    """Run PROPKA software.
+
+    Args:
+        options:  list of PROPKA options
+        pdb_path:  path to PDB file
+        tmp_path:  path for working directory
+    """
+    options += [str(pdb_path)]
+    args = propka.lib.loadOptions(options)
+    try:
+        _LOGGER.warning("Working in tmpdir %s because of PROPKA file output; need to fix this.",
+                        tmp_path)
+        cwd = Path.cwd()
+        os.chdir(tmp_path)
+        molecule = propka.molecular_container.Molecular_container(str(pdb_path), args)
+        molecule.calculate_pka()
+        molecule.write_pka()
+    except Exception as err:
+        raise err
```

Comment:
what's the point of catching and re-raising the exception?

Reply:
Agreed, you can still do the finally.

Reply:
D'oh... I was trying something and forgot to clean up after myself.  Sorry!  Will fix.

Reply:
Fixed in https://github.com/Electrostatics/propka-3.1/commit/1b0ad999b11b2d710d60cf8c14125fbb36e36708